src/auth/index.js

```diff
@@ -11,27 +11,57 @@ const jwtDecode = require('jwt-decode');
 
 const EMBEDDED_USER_AUTH_DATA_PARTS = 4;
 
-export default class Auth {
+export class AuthFactory {
+  constructor() {
+    throw new StitchError('StitchClient can only be made from the StitchClientFactory.create function');
+  }
+
+  static create(client, rootUrl, options) {
+    return new Promise((resolve, reject) => {
+      resolve(new Auth(client, rootUrl, options)._willInitialize);
+    });
+  }
+}
+
+export class Auth {
   constructor(client, rootUrl, options) {
-    let namespace;
-    if (!client || client.clientAppID === '') {
-      namespace = 'admin';
-    } else {
-      namespace = `client.${client.clientAppID}`;
-    }
+    this._willInitialize = new Promise((resolve, reject) => {
+      let namespace;
+      if (!client || client.clientAppID === '') {
+        namespace = 'admin';
+      } else {
+        namespace = `client.${client.clientAppID}`;
+      }
 
-    options = Object.assign({
-      codec: authCommon.APP_CLIENT_CODEC,
-      namespace: namespace,
-      storageType: 'localStorage'
-    }, options);
-
-    this.client = client;
-    this.rootUrl = rootUrl;
-    this.codec = options.codec;
-    this.platform = options.platform || _platform;
-    this.storage = createStorage(options);
-    this.providers = createProviders(this, options);
+      options = Object.assign({
+        codec: authCommon.APP_CLIENT_CODEC,
+        namespace: namespace,
+        storageType: 'localStorage'
+      }, options);
+
+      this.client = client;
+      this.rootUrl = rootUrl;
+      this.codec = options.codec;
+      this.platform = options.platform || _platform;
+      this.storage = createStorage(options);
+      this.providers = createProviders(this, options);
+
+      Promise.all([
+        this._get(),
+        this.storage.get(authCommon.REFRESH_TOKEN_KEY),
+        this.storage.get(authCommon.USER_LOGGED_IN_PT_KEY),
+        this.storage.get(authCommon.DEVICE_ID_KEY)
+      ]).then(([auth, rt, loggedInProviderType, deviceId]) => {
+        this.auth = auth;
+        this.authedId = auth.userId;
+        this.rt = rt;
+        this.loggedInProviderType = loggedInProviderType;
+        this.deviceId = deviceId;
+        this._isInitialized = true;
```

Comment:
[nit] I feel it cleaner to have no `_isInitialized` like variables as I mentioned in previous review. Instead, why not return the `Promise.all`? That way we get rejections to happen correctly.